exporter/opentelemetry-exporter-prometheus-remote-write/tests/test_prometheus_remote_write_exporter.py

```diff
@@ -0,0 +1,337 @@
+            self._generate_record(SumAggregator),
+            self._generate_record(MinMaxSumCountAggregator),
+            self._generate_record(HistogramAggregator),
+            self._generate_record(LastValueAggregator),
+            self._generate_record(ValueObserverAggregator),
+        ]
+        data = self._exporter.convert_to_timeseries(test_records)
+        self.assertEqual(len(data), 5)
+        for timeseries in data:
+            self.assertEqual(timeseries, "test_value")
+
+        no_type_records = [self._generate_record(lambda: None)]
+        with self.assertRaises(ValueError):
+            self._exporter.convert_to_timeseries(no_type_records)
+
+    # Ensures conversion to timeseries fails for unsupported aggregation types
+    def test_invalid_convert_to_timeseries(self):
+        no_type_records = [self._generate_record(lambda: None)]
+        with self.assertRaises(ValueError):
+            self._exporter.convert_to_timeseries(no_type_records)
+
+    # Ensures sum aggregator is correctly converted to timeseries
+    def test_convert_from_sum(self):
+        sum_record = self._generate_record(SumAggregator)
+        sum_record.aggregator.update(3)
+        sum_record.aggregator.update(2)
+        sum_record.aggregator.take_checkpoint()
+
+        self._exporter.create_timeseries = self._converter_mock()
+        timeseries = self._exporter.convert_from_sum(sum_record)
+        self.assertEqual(timeseries[0], (SumAggregator, "testname", 5))
+
+    # Ensures sum min_max_count aggregator is correctly converted to timeseries
+    def test_convert_from_min_max_sum_count(self):
+        min_max_sum_count_record = self._generate_record(
+            MinMaxSumCountAggregator
+        )
+        min_max_sum_count_record.aggregator.update(5)
+        min_max_sum_count_record.aggregator.update(1)
+        min_max_sum_count_record.aggregator.take_checkpoint()
+
+        self._exporter.create_timeseries = self._converter_mock()
+        timeseries = self._exporter.convert_from_min_max_sum_count(
+            min_max_sum_count_record
+        )
+        self.assertEqual(
+            timeseries[0], (MinMaxSumCountAggregator, "testname_min", 1)
+        )
+        self.assertEqual(
+            timeseries[1], (MinMaxSumCountAggregator, "testname_max", 5)
+        )
+        self.assertEqual(
+            timeseries[2], (MinMaxSumCountAggregator, "testname_sum", 6)
+        )
+        self.assertEqual(
+            timeseries[3], (MinMaxSumCountAggregator, "testname_count", 2)
+        )
+
+    # Ensures histogram aggregator is correctly converted to timeseries
+    def test_convert_from_histogram(self):
+        histogram_record = self._generate_record(HistogramAggregator)
+        histogram_record.aggregator.update(5)
+        histogram_record.aggregator.update(2)
+        histogram_record.aggregator.update(-1)
+        histogram_record.aggregator.take_checkpoint()
+
+        self._exporter.create_timeseries = self._converter_mock()
```

Comment:
Same here, an attribute of an object under test is being changed into a `Mock` but it is not reset afterwards.

Reply:
Same as above comment.